CHANGELOG.md

```diff
@@ -18,6 +18,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Bumps `eslint-config-prettier` from 8.8.0 to 9.0.0
 ### Changed
 - Make handling of long numerals an option that is disabled by default ([#557](https://github.com/opensearch-project/opensearch-js/pull/557))
+- Refactor BaseConnectionPool empty method so that callback will always get called even when there is no connection ([#585](https://github.com/opensearch-project/opensearch-js/pull/585)) 
```

Comment:
I think this is really a bug fix, the callback should be called when there's no connections. So I would move this to fixes and say "Fix: ensure connection pool callback with no connections"